keras_cv/layers/preprocessing/random_augmentation_pipeline.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class RandomAugmentationPipeline(
+    tf.keras.__internal__.layers.BaseImageAugmentationLayer
+):
+    """RandomAugmentationPipeline constructs a pipeline based on provided arguments.
+
+    The implemented policy does the following: for each sample provided in `call`(), the
+    policy first samples a random number, if the number is < rate, the policy then
+    selects a random layer from the provided list of `layers`.  It then calls the
+    `layer()` on the sample.  This is done `augmentations_per_image` times.
+
+    This layer can be used to create custom policies resembling `RandAugment` or
+    `AutoAugment`.
+
+    Usage:
+    ```python
+    # construct a list of layers
+    layers = keras_cv.layers.RandAugment.get_standard_policy(
+        value_range=(0, 255), magnitude=0.75, magnitude_stddev=0.3
+    )
+    layers = layers[:4]  # slice out some layers you don't want for whatever reason
+    layers = layers + [keras_cv.layers.GridMask()]
+
+    # create the pipeline.
+    pipeline = keras_cv.layers.RandomAugmentationPipeline(
+        layers=layers, augmentations_per_image=3
+    )
+
+    augmented_images = pipeline(images)
+    ```
+
+    Args:
+        layers: a list of `keras.Layers`.  These are randomly sampled during
```

Comment:
Is there any particular requirement for the layer? eg it need to be BaseImageAugmentationLayer?

Reply:
I think this is nicer and decouple the sublayers with the container layer.

Reply:
Added that these must subclass.  Originally I didn't want to point users this way,  but otherwise we'd have to show them how to augmented the dictionary inputs.  We should probably expose BaseImageAugmentationLayer someday, but not yet.